test/FunctionalTests/Microsoft.OData.Core.Tests/Json/MockStreamBasedJsonWriterFactoryWrapper.cs

```diff
@@ -0,0 +1,57 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MockStreamBasedJsonWriterFactoryWrapper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.OData.Json;
+
+namespace Microsoft.OData.Tests.Json
+{
+    /// <summary>
+    /// This wraps an <see cref="IStreamBasedJsonWriterFactory"/> to
+    /// intercept calls made to create a writer in order to allow
+    /// testing code paths around JSON writer construction.
+    /// </summary>
+    internal class MockStreamBasedJsonWriterFactoryWrapper : IStreamBasedJsonWriterFactory
+    {
+        private readonly IStreamBasedJsonWriterFactory innerFactory;
+
+        public MockStreamBasedJsonWriterFactoryWrapper(IStreamBasedJsonWriterFactory wrappedFactory)
+        {

```

Comment:
```suggestion
        {
            if (wrappedFactory == null)
            {
                throw new ArgumentNullException(nameof(wrappedFactory));
            }
            
```

Reply:
Made the change. Note that this class is only used in tests.